demo/src/test/java/uk/co/ribot/easyadapterdemo/EasyAdapterTest.java

```diff
@@ -0,0 +1,184 @@
+        assertEquals(person2, mEasyAdapter.getItem(1));
+    }
+
+    @Test public void testSetItems() throws Exception {
+        List<Person> list = getSomePeople();
+        assertEquals(0, mEasyAdapter.getItems().size());
+        mEasyAdapter.setItems(list);
+        assertEquals(list, mEasyAdapter.getItems());
+        verify(mMockDataSetObserver).onChanged();
+    }
+
+    @Test public void testSetItemsWithoutNotifying() throws Exception {
+        List<Person> list = getSomePeople();
+        assertEquals(0, mEasyAdapter.getItems().size());
+        mEasyAdapter.setItemsWithoutNotifying(list);
+        assertEquals(list, mEasyAdapter.getItems());
+        verify(mMockDataSetObserver, never()).onChanged();
+    }
+
+    @Test public void testGetItems() throws Exception {
+        List<Person> list = getSomePeople();
+        EasyAdapter<Person> easyAdapter = new EasyAdapter<>(RuntimeEnvironment.application,
+                PersonViewHolder.class, list);
+        assertEquals(list, easyAdapter.getItems());
+    }
+
+    @Test public void testAddItem() throws Exception {
+        mEasyAdapter.getItems().addAll(getSomePeople());
+        Person newPerson = createPerson("New Person");
+        mEasyAdapter.addItem(newPerson);
+
+        assertTrue(mEasyAdapter.getItems().contains(newPerson));
+        verify(mMockDataSetObserver).onChanged();
+    }
+
+    @Test public void testRemoveItem() throws Exception {
+        List<Person> items = getSomePeople();
+        Person personToRemove = items.get(0);
+        mEasyAdapter.getItems().addAll(items);
+        boolean result = mEasyAdapter.removeItem(personToRemove);
+
+        assertTrue(result);
+        assertFalse(mEasyAdapter.getItems().contains(personToRemove));
+        verify(mMockDataSetObserver).onChanged();
+    }
+
+    @Test public void testRemoveNonExistingItem() throws Exception {
+        Person person = createPerson("Person1");
+        boolean result = mEasyAdapter.removeItem(person);
+
+        assertFalse(result);
+        verify(mMockDataSetObserver, never()).onChanged();
+    }
+
+    @Test public void testAddItems() throws Exception {
+        Person person1 = createPerson("Person1");
+        Person person2 = createPerson("Person2");
+        Person person3 = createPerson("Person3");
+        mEasyAdapter.getItems().add(person1);
+        boolean result = mEasyAdapter.addItems(Arrays.asList(person2, person3));
+
+        assertTrue(result);
+        assertEquals(3, mEasyAdapter.getCount());
+        assertTrue(mEasyAdapter.getItems().contains(person2));
+        assertTrue(mEasyAdapter.getItems().contains(person3));
+        verify(mMockDataSetObserver).onChanged();
+    }
+
+    @Test public void testRemoveItems() throws Exception {
+        Person person1 = createPerson("Person1");
+        Person person2 = createPerson("Person2");
+        Person person3 = createPerson("Person3");
+        mEasyAdapter.getItems().addAll(Arrays.asList(person1, person2, person3));
```

Comment:
Because `addItems()` will call `notifyDataSetChanged()` plus this method is not testing  `addItems()` so I was intentionally trying to avoid using it here.